src/components/dragBoxLayer.ts

```diff
@@ -64,37 +65,48 @@ export module Components {
       var resizingEdges: _EdgeIndicator;
       var topLeft: Point;
       var bottomRight: Point;
-      var startedNewBox: boolean;
+      var lastEndPoint: Point;
+
+      var DRAG_MODES = {
+        newBox: 0,
+        resize: 1,
+        move: 2
+      };
+      var mode = DRAG_MODES.newBox;
 
       this._dragInteraction.onDragStart((s: Point) => {
```

Comment:
Let's not abbreviate vars to single letter. Call it `startPoint` or at least `startP`